test/unit/multiplication/test_multiplication_general.cpp

```diff
@@ -212,6 +213,7 @@ TYPED_TEST(GeneralSubMultiplicationDistTestGPU, CorrectnessDistributed) {
       const TypeParam beta = TypeUtilities<TypeParam>::element(-2.6, .7);
       testGeneralSubMultiplication<TypeParam, Backend::GPU, Device::GPU>(comm_grid, a, b, alpha, beta, m,
                                                                          mb);
+      pika::threads::get_thread_manager().wait();
```

Comment:
These `wait`s should go away with a merge of master now that https://github.com/eth-cscs/DLA-Future/pull/844 is merged.